src/PKSim.Core/Extensions/WithDisplayUnitExtensions.cs

```diff
@@ -1,15 +1,23 @@
-﻿using PKSim.Core.Services;
-using OSPSuite.Core.Domain;
+﻿using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.UnitSystem;
+using PKSim.Core.Chart;
 
 namespace PKSim.Core.Extensions
 {
    public static class WithDisplayUnitExtensions
    {
       private static readonly DoubleFormatter _doubleFormatter = new DoubleFormatter();
 
-      public static string DisplayValue(this IWithDisplayUnit withDisplayUnit, double valueInCoreUnit)
+      public static string DisplayValueWithUnit(this IWithDisplayUnit withDisplayUnit, double valueInBaseUnit, IDimension valueDimension = null)
       {
-         return _doubleFormatter.Format(withDisplayUnit.ConvertToDisplayUnit(valueInCoreUnit), withDisplayUnit.DisplayUnit);
+         return _doubleFormatter.Format(DisplayValue(withDisplayUnit, valueInBaseUnit, valueDimension), withDisplayUnit.DisplayUnit);
+      }
+
+      public static double DisplayValue(this IWithDisplayUnit withDisplayUnit, double valueInBaseUnit, IDimension valueDimension = null)
+      {
+         var dimensionToUse = valueDimension ?? withDisplayUnit.Dimension;
+         //This value now needs to be converted in the display value of the target unit
+         return dimensionToUse.BaseUnitValueToUnitValue(withDisplayUnit.DisplayUnit, valueInBaseUnit);
```

Comment:
This is the key here. I am using the dimension to convert to the axis display dimension as opposed to using the dmension of the axis